### ISSUE TITLE
Cannot build on 0.69.21 with C# library

### ISSUE THREAD

**Issue by cjhines** [microsoft/react-native-windows#11345]
### Problem Description

I'm trying to get an older library (react-native-file-viewer) to run in my project.

I made a fork to update dependencies and minimal build which you can find [here](https://github.com/hybridheroes/react-native-file-viewer).

I was running into UAP version errors until I implemented [this workaround](https://github.com/microsoft/react-native-windows/issues/10059#issuecomment-1147939186) of setting an `AssetTargetFallback` property:

```
  <PropertyGroup Label="NuGet">
    <ResolveNuGetPackages>false</ResolveNuGetPackages>
    <AssetTargetFallback>$(AssetTargetFallback);uap10.0.19041</AssetTargetFallback>
  </PropertyGroup>
  ```
  
I'm now getting the following error when attempting a build on CI:

```
 C:\MyApp\application\node_modules\react-native-windows\Microsoft.ReactNative.Managed.CodeGen\target\x64\Release\Microsoft.ReactNative.Managed.CodeGen\publish\Microsoft.ReactNative.Managed.CodeGen.exe @C:\MyApp\application\node_modules\react-native-file-viewer\windows\ReactNativeFileViewer\obj\x64\Release\ReactNativeFileViewer.ReactNativeCodeGenReactNativeFileViewer.ReactNativeCodeGen.rsp
  Invalid runtimeconfig.json [C:\MyApp\application\node_modules\react-native-windows\Microsoft.ReactNative.Managed.CodeGen\target\x64\Release\Microsoft.ReactNative.Managed.CodeGen\publish\Microsoft.ReactNative.Managed.CodeGen.runtimeconfig.json] [C:\MyApp\application\node_modules\react-native-windows\Microsoft.ReactNative.Managed.CodeGen\target\x64\Release\Microsoft.ReactNative.Managed.CodeGen\publish\Microsoft.ReactNative.Managed.CodeGen.runtimeconfig.dev.json]
C:\MyApp\application\node_modules\react-native-windows\PropertySheets\ManagedCodeGen\Microsoft.ReactNative.Managed.CodeGen.targets(78,5): error MSB3073: The command "C:\MyApp\application\node_modules\react-native-windows\Microsoft.ReactNative.Managed.CodeGen\target\x64\Release\Microsoft.ReactNative.Managed.CodeGen\publish\Microsoft.ReactNative.Managed.CodeGen.exe @C:\MyApp\application\node_modules\react-native-file-viewer\windows\ReactNativeFileViewer\obj\x64\Release\ReactNativeFileViewer.ReactNativeCodeGenReactNativeFileViewer.ReactNativeCodeGen.rsp" exited with code -2147450733. [C:\MyApp\application\node_modules\react-native-file-viewer\windows\ReactNativeFileViewer\ReactNativeFileViewer.csproj]
```

I can't find any reference to this error outside of dotnet forums and I'm struggling to apply their use cases to mine.

I thought it could be due to the long paths, but this should be allowed:

<img width="427" alt="Screenshot 2023-03-07 at 21 12 54" src="https://user-images.githubusercontent.com/16542421/223541646-f5925554-2c4f-4530-8305-016b35dbbc09.png">

### Steps To Reproduce

1. Install my [react-native-file-viewer fork](https://github.com/hybridheroes/react-native-file-viewer).
2. Follow the workaround [outlined here](https://github.com/microsoft/react-native-windows/issues/10059#issuecomment-1147939186)

### CLI version

npx react-native --version

### Environment

```markdown
System:
    OS: Windows 10 10.0.20348
    CPU: (4) x64 Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz
    Memory: 11.96 GB / 16.00 GB
  Binaries:
    Node: 14.20.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.17 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.11.32630.194 (Visual Studio Community 2019)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.7 => 0.69.7
    react-native-windows: 0.69.6 => 0.69.6
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Platform Version

None

### Visual Studio Version

Visual Studio 2019

### Build Configuration

None

**Comment by chrisglein** [comment]
Title of your issues says C++ library, but the implementation of that module [looks like it's C#](https://github.com/hybridheroes/react-native-file-viewer/blob/master/windows/ReactNativeFileViewer/ReactNativeModule.cs). Typo?

This failure looks to be in the codegen. There was a change in 0.71 that updates the dotNet version that could affect this. Is upgrading an option?

Also 0.69.6 is a bit behind. 0.69.21 is the latest (https://github.com/microsoft/react-native-windows/releases).

Anyway, not confident this has been patched. But it'll help in investigation if we're dealing with latest. There has been motion in this area.

**Comment by cjhines** [comment]
@chrisglein Apologies, it is indeed a C# module. I've updated the title.

I tried 0.69.21 with no luck unfortunately.

> Is upgrading an option?

Not in the short term unfortunately. Might look into it as a last resort but it would affect our release schedule.

> How is it being built in CI?

Here is the build command I'm using:

`npx react-native run-windows --logging --no-launch --no-deploy --release --msbuildprops PackageCertificateKeyFile=keyfile.pfx,PackageCertificatePassword="${CERT_PW}"`
